lib/engine/game/g_1848/step/special_track.rb

```diff
@@ -8,12 +8,13 @@ module G1848
       module Step
         class SpecialTrack < Engine::Step::SpecialTrack
           def process_lay_tile(action)
-            ability = abilities(action.entity)
             super
+            # track lay ability is both on player and corporate, remove all left over abilities
+            company = action.entity
+            company.all_abilities.each { |ab| company.remove_ability(ab) }
             return unless @game.private_closed_triggered
```

Comment:
If you are going to do this pattern multiple times, you can create a module to handle.

Reply:
do you have a working example on how to do this?

Reply:
The InterestOnLoans module is a good example of functionality being encapsulated into a module. In your case, you may have a module with a method with the signature ability_used!(company). This could also be a game method instead, if you don't want to go the module way. Half a dozen of one, 6 of another.

Also, all_abilities.each { || company.remove_ability() } will not work the way you want. You can't iterate over a list of abilities that you are removing items from. Instead, use all_abilities.dup.each.